action-scripts/translate.py

```diff
@@ -1,15 +1,24 @@
+from typing import Optional
 from translate_scripts import build_pot, build_trans_dict, compile_po_to_mo
-from bpy_addon_build.api import BabContext
+from bpy_addon_build.api import BabContext, BpyError
 import sys
 
-def pre_build(ctx: BabContext) -> None:
+def pre_build(ctx: BabContext) -> Optional[BpyError]:
     sys.stdout.reconfigure(encoding='utf-8')
-    build_pot.pre_build(ctx)
+    return build_pot.pre_build(ctx)
 
 def main():
     sys.stdout.reconfigure(encoding='utf-8')
-    compile_po_to_mo.main()
-    build_trans_dict.main()
+    
+    # Go through each module, and 
+    # run its main function, while checking 
+    # for a return value
+    modules = [compile_po_to_mo, build_trans_dict]
+    for mod in modules:
+        res = mod.main()
+        if res:
+            continue 
+        return res
```

Comment:
From reading the code on its own, it's not very clear to me _why_ we continue. Does a return value imply an error or not? If so, maybe this is more readable while also being shorter


```suggestion
        error = mod.main()
        if error:
            return error
```